### PR TITLE
Fix/auth

### PR DESCRIPTION
**Pull request by AlexYalovoy**

### CODE REVIEW

File: src/pages/SignUp/index.js [hunk 1]
```diff
@@ -1,54 +1,30 @@
...
-      isInvalid={isInvalid}
-      onSubmit={this.onSubmit}
+      onSubmit={onSubmit}
       error={error}
     />
   );
-  }
 }
 
 const mapStateToProps = (state) => {
```
Comment (motifej) [inline src/pages/SignUp/index.js]: const mapStateToProps = ({ signUp }) => ({ error: signUp.error })

Comment (motifej) [reply src/pages/SignUp/index.js]: rename signUpReducer to signUp

File: src/pages/SignUp/index.js [hunk 2]
```diff
@@ -1,54 +1,30 @@
...
 import { signUp } from '../../actions/signUp';
 
-class SignUpFormBase extends Component {
+const SignUpFormBase = (props) => {
   
-  onSubmit = event => {
-    this.props.signUp(this.props.email, this.props.passwordOne);
-    event.preventDefault();
+  function onSubmit(...params) {
+    const {email, passwordOne} = params[0];
```
Comment (motifej) [inline src/pages/SignUp/index.js]: fix spaces

Comment (motifej) [reply src/pages/SignUp/index.js]: why do you use rest?

File: src/pages/SignUp/index.js [hunk 3]
```diff
@@ -1,54 +1,30 @@
...
 import { compose } from 'recompose';
-import {SignUp} from '../../components'
-import {connect} from 'react-redux';
+import { SignUp } from '../../components'
+import { connect } from 'react-redux';
 import { withFirebase } from '../../api/firebase';
 import { signUp } from '../../actions/signUp';
 
-class SignUpFormBase extends Component {
+const SignUpFormBase = (props) => {
```
Comment (motifej) [inline src/pages/SignUp/index.js]: use destructure

File: src/components/SignUp/index.js [hunk 4]
```diff
@@ -1,64 +1,57 @@
...
+import Phone from '@material-ui/icons/Phone';
 
-import AccountIcon from "@material-ui/icons/AccountCircle";
-import MailIcon from "@material-ui/icons/Mail";
-import Security from "@material-ui/icons/Security";
-import Phone from "@material-ui/icons/Phone";
+import Input from '../Input';
+import styles from './style.js';
+import {
+  validateFirstName,
```
Comment (motifej) [inline src/components/SignUp/index.js]: left only one for text input

File: src/pages/SignUp/index.js [hunk 5]
```diff
@@ -6,15 +6,13 @@ import { connect } from 'react-redux';
 import { withFirebase } from '../../api/firebase';
 import { signUp } from '../../actions/signUp';
 
-const SignUpFormBase = (props) => {
+const SignUpFormBase = ({error, signUp}) => {
   
-  function onSubmit(...params) {
-    const {email, passwordOne} = params[0];
-    props.signUp(email, passwordOne);
+  const onSubmit = (inputs) => {
```
Comment (AndrZakhara) [inline src/pages/SignUp/index.js]: Use arrow function with one argument without brackets. In this syntax, we will have a warning from eslint .

File: src/pages/SignUp/index.js [hunk 6]
```diff
@@ -1,57 +1,27 @@
...
-  return {
-    email: state.form.signUp && state.form.signUp.values && state.form.signUp.values.email,
-    passwordOne: state.form.signUp && state.form.signUp.values && state.form.signUp.values.passwordOne,
-    passwordTwo: state.form.signUp && state.form.signUp.values && state.form.signUp.values.passwordTwo,
-    username: state.form.signUp && state.form.signUp.values && state.form.signUp.values.username,
-    phone: state.form.signUp && state.form.signUp.values && state.form.signUp.values.phone,
-    error: state.signUpReducer.error
-  }
-}
+const mapStateToProps = ({signUp}) => ({error: signUp.error});
```
Comment (motifej) [inline src/pages/SignUp/index.js]: fix spaces

File: src/reducers/signUp/index.js [hunk 7]
```diff
@@ -27,6 +27,7 @@ export default (state = initialState, { type, payload }) => {
       };
     case SIGNUP_ERROR:
       console.log(SIGNUP_ERROR);
+      console.log(payload);
```
Comment (motifej) [inline src/reducers/signUp/index.js]: remove console

File: src/sagas/signUpSaga/index.js [hunk 8]
```diff
@@ -1,15 +1,15 @@
...
+import { signUpError,signUpSuccess } from '../../actions/signUp';
+import { register } from '../../api/auth';
 
 function* SignUp(action) {   
    const { email, password } = action.payload;
    try {
-      yield Firebase.DoCreateUserWithEmailAndPassword(email, password)
-      yield put({type: SIGNUP_SUCCESS});
+      yield call(register, email, password);
+      yield put(call(signUpSuccess));
```
Comment (motifej) [inline src/sagas/signUpSaga/index.js]: call - just for async function

File: src/pages/SignUp/index.js [hunk 9]
```diff
@@ -1,68 +1,31 @@
...
-    username: state.form.signUp && state.form.signUp.values && state.form.signUp.values.username,
-    phone: state.form.signUp && state.form.signUp.values && state.form.signUp.values.phone,
-    error: state.signUpReducer.error
-  }
-}
+const mapStateToProps = ({ signUp }) => ({ error: signUp.error });
 
 const mapDispatchToProps = dispatch => ({
-  signUp: (email, password) => dispatch(signUp(email, password)),
+  signUpDispatch: (email, password) => dispatch(signUp(email, password)),
```
Comment (motifej) [inline src/pages/SignUp/index.js]: too bad!!! should be like
const mapDispatchToProps = {
  signUp,
}

**Pull request merged**